### DISCUSSION TITLE
Pretty URLs

### DISCUSSION THREAD

**Discussion by yestool** [rochacbruno/marmite#114] [General]
how to set pretty urls? 

current:
  result: site/about.html   --->  https://www.myweb.com/about.html
want:
  result: site/about/index.html  https://www.myweb.com/about/

**Comment by rochacbruno** [comment]
There is no support for this, this would require creating subfolders for each content with an index.html inside it.

I don't want it because it will not be a flat site anymore, will not be possible to just open the site on the browser without the need of web server and will require URL resolution.

I understand that some people prefer URLs without HTML in it, it can still be achieved if you configure your web server (nginx) to resolve it but is not a marmite feature.

Some of the design decisions are based on generated site being flat html.

- easy to refer media files `./media/file.png`
- back links collection on pattern `./{slug}.html`
- easy to link other pages `./slug.html`
- easy to link static files `./static/style.css`

So to implement subfolders, a shortcode on markdown would need to be provided to resolve URLs, this would require doing some pre-processing of markdown, build the URLs considering a base_path if site is not deployed to the root of domain.
 
I don't think we will go this path right now (unless someone submits a PR that doesn't break the flat site experience)

We already have Zola and Cobalt that are able to solve this.

anyway, Thanks for reporting.

**Comment by mandarvaze** [comment]
> how to set pretty urls?
> 
> current: result: site/about.html ---> https://www.myweb.com/about.html 
> want: result: site/about/index.html https://www.myweb.com/about/

This might be useful for people wanting to migrate from hugo.

**Reply by mandarvaze** [reply]
On more digging, I found that even in hugo one can use `uglyURLs` via [this config](https://gohugo.io/configuration/ugly-urls/) change.

**Comment by mandarvaze** [comment]
In case you are deploying on Netlify, [this feature](https://docs.netlify.com/build/post-processing/overview/) may be useful.
Of course, Netlify officially does not support `marmite` but one can [request support](https://github.com/rochacbruno/marmite/discussions/369#discussioncomment-15346032)
